.github/workflows/ci/pkg/environment/environment.go

```diff
@@ -0,0 +1,228 @@
+package environment
+
+import (
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"os"
+
+	"github.com/gravitational/teleport/.github/workflows/ci"
+	"github.com/gravitational/trace"
+
+	"github.com/google/go-github/v37/github"
+)
+
+// Config is used to configure Environment
+type Config struct {
+	Client        *github.Client
+	Reviewers     string
+	EventPath     string
+	Token         string
+	unmarshalRevs unmarshalReviewersFn
+}
+
+// Environment contains information about the environment
+type Environment struct {
+	Client           *github.Client
+	PullRequest      *PullRequestMetadata
+	token            string
+	reviewers        map[string][]string
+	defaultReviewers []string
+	action           string
+}
+
+type unmarshalReviewersFn func(str string, client *github.Client) (map[string][]string, error)
+
+// CheckAndSetDefaults verifies configuration and sets defaults
+func (c *Config) CheckAndSetDefaults() error {
+	if c.Client == nil {
+		return trace.BadParameter("missing parameter Client.")
+	}
+	if c.Reviewers == "" {
+		return trace.BadParameter("missing parameter Reviewers.")
+	}
+	if c.EventPath == "" {
+		return trace.BadParameter("missing parameter EventPath.")
+	}
+	if c.Token == "" {
+		return trace.BadParameter("missing parameter token.")
+	}
+	if c.unmarshalRevs == nil {
+		c.unmarshalRevs = unmarshalReviewers
+	}
+	return nil
+}
+
+// New creates a new instance of Environment.
+func New(c Config) (*Environment, error) {
+	var env Environment
+	err := c.CheckAndSetDefaults()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	env.Client = c.Client
+	revs, err := c.unmarshalRevs(c.Reviewers, c.Client)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	env.reviewers = revs
+	env.defaultReviewers = revs[""]
+	err = env.SetPullRequest(c.EventPath)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return &env, nil
+}
+
+// unmarshalReviewers converts the passed in string representing json object into a map
+func unmarshalReviewers(str string, client *github.Client) (map[string][]string, error) {
+	var hasDefaultReviewers bool
+	if str == "" {
+		return nil, trace.BadParameter("reviewers not found.")
+	}
+	m := make(map[string][]string)
+
+	err := json.Unmarshal([]byte(str), &m)
+	if err != nil {
+		return nil, err
+	}
+	for k, v := range m {
+		for _, reviewer := range v {
+			err := userExists(reviewer, client)
+			if err != nil {
+				return nil, trace.Wrap(err)
+			}
+		}
+		if k == "" {
+			hasDefaultReviewers = true
+			continue
+		}
+		err := userExists(k, client)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+	}
+	if !hasDefaultReviewers {
+		return nil, trace.BadParameter("default reviewers are not set. set default reviewers with an empty string as a key.")
+	}
+	return m, nil
+
+}
+
+// userExists checks if a user exists
+func userExists(user string, client *github.Client) error {
```

Comment:
Based on the function name, it seems like it should either return a boolean, or (user, error) tuple.